### DISCUSSION TITLE
How to handle connections when scaling redis instances?

### DISCUSSION THREAD

**Discussion by italopenaforte** [Grokzen/redis-py-cluster#504] [Q&A]
In the process of scaling redis, services end up having a lot of connection timeout or MOVED ERROR.

How to deal with the connection to not need to restart services after redis scaling happens?

**Comment by Grokzen** [comment]
@italopenaforte I dont know really. Usually the connection pool deals with this automatically in some way shape or form :shrug:

It is impossible to not have MOVED ERROR when you rescale or modify and the connection pool will automatically sort it out and finding the new host and chug along eventually.

**Reply by italopenaforte** [reply]
I understand that it is impossible not to get the MOVED ERROR, but there is something that can be done to "reset" the connection without having to restart the services?

**Reply by Grokzen** [reply]
If you have access to the cluster objects then you can just run the `.reset()` method on the ConnectionPool found here https://github.com/Grokzen/redis-py-cluster/blob/master/rediscluster/connection.py#L183 . If you do not have any programatic access to the objects while they are running then you are out of luck

**Reply by italopenaforte** [reply]
hahahahah Thanks a lot for the help, I'll take a closer look at this point.
